common/lib/xmodule/xmodule/tests/test_sequence.py

```diff
@@ -0,0 +1,148 @@
+        """
+        Implements abstract method for getting the current user.
+        """
+        user = XBlockUser()
+        user.opt_attrs['edx-platform.username'] = 'test user'
+        return user
+
+
+class SequenceBlockTestCase(XModuleXmlImportTest):
+    """
+    Tests for the Sequence Module.
+    """
+    def setUp(self):
+        super(SequenceBlockTestCase, self).setUp()
+
+        course_xml = self._set_up_course_xml()
+        self.course = self.process_xml(course_xml)
+        self._set_up_module_system(self.course)
+
+        for chapter_index in range(len(self.course.get_children())):
+            chapter = self._set_up_block(self.course, chapter_index)
+            setattr(self, 'chapter_{}'.format(chapter_index + 1), chapter)
+
+            for sequence_index in range(len(chapter.get_children())):
+                sequence = self._set_up_block(chapter, sequence_index)
+                setattr(self, 'sequence_{}_{}'.format(chapter_index + 1, sequence_index + 1), sequence)
+
+    def _set_up_course_xml(self):
+        """
+        Sets up and returns XML course structure.
+        """
+        course = xml.CourseFactory.build()
+
+        chapter_1 = xml.ChapterFactory.build(parent=course)  # has 2 child sequences
+        xml.ChapterFactory.build(parent=course)  # has 0 child sequences
+        chapter_3 = xml.ChapterFactory.build(parent=course)  # has 1 child sequence
+        chapter_4 = xml.ChapterFactory.build(parent=course)  # has 2 child sequences
+
+        xml.SequenceFactory.build(parent=chapter_1)
+        xml.SequenceFactory.build(parent=chapter_1)
+        sequence_3_1 = xml.SequenceFactory.build(parent=chapter_3)  # has 3 verticals
+        xml.SequenceFactory.build(parent=chapter_4)
+        xml.SequenceFactory.build(parent=chapter_4)
+
+        for _ in range(3):
+            xml.VerticalFactory.build(parent=sequence_3_1)
+
+        return course
+
+    def _set_up_block(self, parent, index_in_parent):
+        """
+        Sets up the stub sequence module for testing.
+        """
+        block = parent.get_children()[index_in_parent]
+
+        self._set_up_module_system(block)
+
+        block.xmodule_runtime._services['bookmarks'] = Mock()  # pylint: disable=protected-access
+        block.xmodule_runtime._services['user'] = StubUserService()  # pylint: disable=protected-access
+        block.xmodule_runtime.xmodule_instance = getattr(block, '_xmodule', None)  # pylint: disable=protected-access
+        block.parent = parent.location
+        return block
+
+    def _set_up_module_system(self, block):
+        """
+        Sets up the test module system for the given block.
+        """
+        module_system = get_test_system()
+        module_system.descriptor_runtime = block._runtime  # pylint: disable=protected-access
+        block.xmodule_runtime = module_system
+
+    def test_render_student_view(self):
+        html = self.assert_student_view_response(self.sequence_3_1, 1, None)
```

Comment:
This might be buried too high up the test hierarchy to fix easily, but this method both performs actions (getting a response from a view) and makes assertions (of an unspecified nature) against those actions.  These things should generally remain separate.  If it's easy to modify this so that the tests take the structure of "I perform action X," and then "I assert that the results of this action are Y," it will improve clarity in the tests.